src/python/pants/bin/local_pants_runner.py

```diff
@@ -271,9 +272,9 @@ def _maybe_run_v2(self):
     except GracefulTerminationException as e:
       logger.debug('Encountered graceful termination exception {}; exiting'.format(e))
       return e.exit_code
-    except Exception as e:
-      logger.warn('Encountered unhandled exception {} during rule execution!'
-                  .format(e))
+    except Exception:
+      logger.warning('Encountered unhandled exception during rule execution!')
+      logger.warning(traceback.format_exc())
```

Comment:
*not really relevant note* Was this changed to `logger.warning()` instead of `.warn()` because of a deprecation warning when using `logger.warn()`? Because I thought I saw that happen elsewhere, too, and it seems like the fact that `logger.warn{,ing}()` are used in `except` blocks means those might not always show up when running pants. I might do a `s/logger\.warn/logger.warning/g` on the repo in another PR.

Reply:
Correct - `warn()` is deprecated.